lib/ss_light.go

```diff
@@ -301,29 +319,17 @@ func (l *LightClient) Start(
 				case <-ctx.Done():
 					return
 				case <-time.After(time.Second * 30):
-					// Allow 30 seconds to see if new leaders were added
-					newMtdt, err := getInfo(sharable, metadata.Cookie.Id, l.pubKey)
-					if err == nil && len(newMtdt.Cookie.Leaders) > count {
-						log.Infof("Got %d new leaders", len(newMtdt.Cookie.Leaders)-count)
-						newLeadersToAdd := []peer.AddrInfo{}
-						for _, v := range newMtdt.Cookie.Leaders {
-							found := false
-							for _, x := range metadata.Cookie.Leaders {
-								if x.ID == v.ID {
-									found = true
-									break
-								}
-							}
-							if !found {
-								newLeadersToAdd = append(newLeadersToAdd, v)
-							}
-						}
-						count += lite.Bootstrap(newLeadersToAdd)
-					}
 					if time.Since(start) > time.Minute*15 {
 						log.Warn("Tried getting more peers for 15mins")
+						showStep(timeoutError, "Download timed out", l.jsonOut)
 						return
 					}
+					// Try to re-bootstrap if client was unable to bootstrap previously
+					if count < len(metadata.Cookie.Leaders) {
+						count += lite.Bootstrap(metadata.Cookie.Leaders)
```

Comment:
This doesnt seem right. If you connect to 2/5 nodes. But then you try to connect to 5 again. Say you only connect to 2 again but its the same 2 nodes. In which case your count will not be correct.

This needs to be count = bootstrap()